File: Tyche.iOS/DataPool/Sources/DataPool/Domain/PoolGamblerScoreRepository.swift
import Core

public protocol PoolGamblerScoreRepository {
    func getPoolGamblerScoresByGambler(
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerScore>, Error>
    
    func getPoolGamblerScoresByPool(
        poolId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerScore>, Error>
    
    func getPoolGamblerScore(
        poolId: String,
        gamblerId: String
    ) async -> Result<PoolGamblerScore, Error>
}

File: Tyche.Android/data/pool/src/main/java/com/felipearpa/data/pool/domain/PoolGamblerScoreRepository.kt
package com.felipearpa.data.pool.domain

import com.felipearpa.tyche.core.paging.CursorPage

interface PoolGamblerScoreRepository {
    suspend fun getPoolGamblerScoresByGambler(
        gamblerId: String,
        next: String? = null,
        searchText: String? = null
    ): Result<CursorPage<PoolGamblerScore>>

    suspend fun getPoolGamblerScoresByPool(
        poolId: String,
        next: String? = null,
        searchText: String? = null
    ): Result<CursorPage<PoolGamblerScore>>

    suspend fun getPoolGamblerScore(poolId: String, gamblerId: String): Result<PoolGamblerScore>
}